### PR TITLE
CLI Testsuite Refactoring

### PR DESCRIPTION
**Pull request by Gonthim**
Further refactoring to get all of the CLI tests using the structure to follow.

### CODE REVIEW

File: cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java [hunk 1]
```diff
@@ -1641,6 +1641,11 @@ private void dohandle(Callback[] callbacks) throws IOException, UnsupportedCallb
                             password = temp.toCharArray();
                         }
                     }
+                    try {
+                        Thread.sleep(5000);
```
Comment (ctomc) [inline cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java]: 5 second sleep? that sounds excessive.

Comment (Gonthim) [inline cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java]: Excessive indeed. Was only for testing.

File: testsuite/shared/src/main/java/org/jboss/as/test/integration/management/cli/CliProcessWrapper.java [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+        InputStream cliStream = null;
+
+        public CliErrorsReader(InputStream inputStream){
+            cliStream = inputStream;
+        }
+
+        @Override
+        public void run() {
+            java.io.InputStreamReader isr = new java.io.InputStreamReader(cliStream);
+            java.io.BufferedReader br = new java.io.BufferedReader(isr);
```
Comment (ctomc) [inline testsuite/shared/src/main/java/org/jboss/as/test/integration/management/cli/CliProcessWrapper.java]: this two are never closed,
probably best if you do add it to try-with-resoruces block

File: testsuite/shared/src/main/java/org/jboss/as/test/integration/management/cli/CliProcessWrapper.java [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+        }
+
+    }
+
+    private void logErrors() {
+        //System.out.println("Failed to execute '" + cmd + "'");
+        System.out.println("Command's output: '" + cliOutputBuffer + "'");
+
+        java.io.InputStreamReader isr = new java.io.InputStreamReader(cliProcess.getErrorStream());
+        java.io.BufferedReader br = new java.io.BufferedReader(isr);
```
Comment (ctomc) [inline testsuite/shared/src/main/java/org/jboss/as/test/integration/management/cli/CliProcessWrapper.java]: same here, readers not closed.

**Comment by bstansberry** [comment]
Superceded by #959, which is merged.

**Pull request closed without merging**